### DISCUSSION TITLE
Box component and its boxSystemProps

### DISCUSSION THREAD

**Discussion by vitorleonel** [system-ui/theme-ui#2286] [General]
Hey,

I noticed that now a days we have a list of valid properties inside the Box component and before we use the `rest` variable to spread the properties to the Component we have a foreach removing the invalid ones.

Does anybody knows if we have a way or a pattern to use custom properties in this case?

https://github.com/system-ui/theme-ui/blob/develop/packages/components/src/Box.tsx

Thank you in advance.

**Comment by lachlanjc** [comment]
Could you be more specific about "custom properties"? Do you mean creating your own components wrapping Box with React props, or CSS custom properties?

**Reply by vitorleonel** [reply]
I mean, how can I use properties like "data-analytics-id"? I saw that we have a fixed list allowing only the properties inside that list.

**Reply by lachlanjc** [reply]
All props you pass except [`BoxOwnProps`](https://github.com/system-ui/theme-ui/blob/develop/packages/components/src/Box.tsx#L59) will be rendered in your HTML, so you can pass `data-*`, `aria-*` etc as usual & they should work normally!
